File: csharp/functions/ClassExample.cs
using System;

class Person
{
    public string Name { get; set; }
    public int Age { get; set; }

    public Person(string name, int age)
    {
        Name = name;
        Age = age;
    }

    public string Greet()
    {
        return $"Hello, I'm {Name} and I'm {Age} years old";
    }

    public void Birthday()
    {
        Age++;
        Console.WriteLine($"Happy birthday! Now {Age} years old");
    }

    public override string ToString()
    {
        return $"Person(name={Name}, age={Age})";
    }
}

class Employee : Person
{
    public string JobTitle { get; set; }

    public Employee(string name, int age, string jobTitle) : base(name, age)
    {
        JobTitle = jobTitle;
    }

    public string Work()
    {
        return $"{Name} is working as a {JobTitle}";
    }

    public override string ToString()
    {
        return $"Employee(name={Name}, age={Age}, job={JobTitle})";
    }
}

class ClassExample
{
    static void Main()
    {
        Person person = new Person("Alice", 30);
        Console.WriteLine(person.Greet());
        person.Birthday();
        Console.WriteLine(person);

        Console.WriteLine();

        Employee employee = new Employee("Bob", 25, "Software Engineer");
        Console.WriteLine(employee.Greet());
        Console.WriteLine(employee.Work());
        Console.WriteLine(employee);
    }
}

File: ruby/functions/class_example.rb
#!/usr/bin/env ruby

class Person
  attr_accessor :name, :age

  def initialize(name, age)
    @name = name
    @age = age
  end

  def greet
    "Hello, I'm #{@name} and I'm #{@age} years old"
  end

  def birthday
    @age += 1
    "Happy birthday! Now #{@age} years old"
  end

  def to_s
    "Person(name=#{@name}, age=#{@age})"
  end
end

class Employee < Person
  attr_accessor :job_title

  def initialize(name, age, job_title)
    super(name, age)
    @job_title = job_title
  end

  def work
    "#{@name} is working as a #{@job_title}"
  end

  def to_s
    "Employee(name=#{@name}, age=#{@age}, job=#{@job_title})"
  end
end

person = Person.new('Alice', 30)
puts person.greet
puts person.birthday
puts person

employee = Employee.new('Bob', 25, 'Software Engineer')
puts employee.greet
puts employee.work
puts employee
